testing/trino-testing-containers/src/main/java/io/trino/testing/containers/PlatformChecks.java

```diff
@@ -0,0 +1,94 @@
+ * limitations under the License.
+ */
+package io.trino.testing.containers;
+
+import com.github.dockerjava.api.DockerClient;
+import com.github.dockerjava.api.command.PullImageResultCallback;
+import com.github.dockerjava.api.model.PullResponseItem;
+import org.testcontainers.DockerClientFactory;
+import org.testcontainers.utility.DockerImageName;
+import org.testcontainers.utility.ImageNameSubstitutor;
+
+import static com.google.common.base.Strings.padEnd;
+import static com.sun.jna.Platform.isARM;
+import static java.lang.System.exit;
+import static java.lang.System.getenv;
+import static java.util.Locale.ENGLISH;
+
+public class PlatformChecks
+        extends ImageNameSubstitutor
+{
+    private static final boolean TESTCONTAINERS_SKIP_ARCH_CHECK = "true".equalsIgnoreCase(getenv("TESTCONTAINERS_SKIP_ARCHITECTURE_CHECK"));
+
+    @Override
+    public DockerImageName apply(DockerImageName dockerImageName)
+    {
+        if (TESTCONTAINERS_SKIP_ARCH_CHECK) {
+            return dockerImageName;
+        }
+
+        DockerClient client = DockerClientFactory.lazyClient();
+        try {
+            client.pullImageCmd(dockerImageName.asCanonicalNameString()).exec(new PullImageResultCallback() {
+                @Override
+                public void onNext(PullResponseItem item)
+                {
+                    String progress = item.getProgress();
+                    if (progress != null) {
+                        System.err.println(padEnd(dockerImageName.asCanonicalNameString() + ":" + item.getId(), 50, ' ') + ' ' + progress);
+                    }
+                }
+            }).awaitCompletion();
+        }
+        catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new RuntimeException("Could not pull image", e);
+        }
+
+        String imageArch = client.inspectImageCmd(dockerImageName.asCanonicalNameString())
+                .exec()
+                .getArch()
+                .toLowerCase(ENGLISH);
+
+        boolean isJavaOnArm = isARM();
+        boolean isImageArmBased = imageArch.contains("arm");
+        boolean hasIncompatibleRuntime = (isJavaOnArm != isImageArmBased);
+
+        if (hasIncompatibleRuntime) {
+            String dockerArch = client.versionCmd().exec().getArch();
+
+            System.err.println("""
+
+                    !!! WARNING !!!
+                    Detected incompatible Docker image and host architectures. The performance of running docker images in such scenarios can vary or not work at all.
+                    Host: %s (%s).
+                    Docker architecture: %s.
+                    Docker image architecture: %s.
+
+                    Set environment variable TESTCONTAINERS_SKIP_ARCHITECTURE_CHECK=true to skip this check.
+                    !!! WARNING !!!
+                    """.formatted(System.getProperty("os.name"), System.getProperty("os.arch"), dockerArch, imageArch));
+
+            exit(99);
```

Comment:
It's not a warning if it exits, but an error, right?

Reply:
Correct

Reply:
Can you then change `!!! WARNING !!!` to `!!! ERROR !!!`?